gcc/testsuite/rust/compile/let-pattern.rs

```diff
@@ -0,0 +1,3 @@
+fn main() {
+    let ((_x, _y, _, _z)) = (1, 2, 3, 4);
+}
```

Comment:
So this is a pattern in parentheses and a tuple pattern at once, right? I think it'd be worth it to separate them into two testcases (and maybe keep this one which contains both) so it'd be easier to pinpoint if eventually something fails.